extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -444,58 +436,33 @@ export class AmpStoryReaction extends AMP.BaseElement {
       });
   }
 
-  /**
-   * Update the Reaction data in the datastore
-   *
-   * @param {number} reactionValue
-   * @private
-   */
-  updateReactionData_(reactionValue) {
-    this.executeReactionRequest_(
-      dict({
-        'method': 'POST',
-      }),
-      reactionValue
-    ).catch((error) => {
-      dev().error(TAG, error);
-    });
-  }
-
   /**
    * Executes a Reactions API call.
    *
-   * @param {Object} requestOptions
+   * @param {string} method GET or POST.
    * @param {number=} reactionValue
-   * @return {Promise<ReactionResponseType|undefined>}
+   * @return {Promise<ReactionResponseType>}
    * @private
    */
-  executeReactionRequest_(requestOptions, reactionValue) {
-    if (!assertAbsoluteHttpOrHttpsUrl(this.element.getAttribute('endpoint'))) {
+  executeReactionRequest_(method, reactionValue) {
+    let url = this.element.getAttribute('endpoint');
+    if (!assertAbsoluteHttpOrHttpsUrl(url)) {
       return Promise.reject(ENDPOINT_INVALID_ERROR);
     }
 
-    if (this.reactionId_ === null) {
-      const pageId = closest(dev().assertElement(this.element), (el) => {
-        return el.tagName.toLowerCase() === 'amp-story-page';
-      }).getAttribute('id');
-
-      this.reactionId_ = `CANONICAL_URL#page=${pageId}`;
-    }
-
-    const requestVars = dict({
-      'reactionType': this.reactionType_,
-      'reactionId': this.reactionId_,
-    });
-
-    let url = this.element.getAttribute('endpoint');
-
     return this.getClientId_().then((clientId) => {
-      requestVars['clientId'] = clientId;
+      const requestOptions = {'method': method};
+      const requestParams = dict({
+        'reactionType': this.reactionType_,
+        'clientId': clientId,
+      });
+      url = appendPathToUrl(url, this.reactionId_);
```

Comment:
This breaks the URL if the publisher specified query params, which makes it impossible to use the AMP variable substitution.
https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md

Reply:
Can fix it so that it adds the path to the url before query and fragment. Expect update on url.js

Reply:
Added support for query params and frament on url.js, and added tests for it. LMK if this works